cargo-shuttle/src/lib.rs

```diff
@@ -901,14 +901,51 @@ impl Shuttle {
     }
 
     async fn deploy(&self, client: &Client, args: DeployArgs) -> Result<CommandOutcome> {
-        if !args.allow_dirty {
-            self.is_dirty()?;
+        let working_directory = self.ctx.working_directory();
+
+        let mut deploy_options: HashMap<String, String> = HashMap::new();
```

Comment:
I think we can omit this type annotation if we use `to_string` in line 922 instead of `into`.

Reply:
> I prefer the explicit annotation in this case.
Really? I try to always avoid them. Taste is different.

Reply:
> However, I think we should consider using a serde struct instead of a HashMap ...

Type safety is always good! We could also use a "simple" (no extractor) struct like this:

``` rust
struct DeployOptions {
    git_branch: Option<String>,
    git_dirty: bool,
    ...
}
```